src/components/ConnectToNetSuiteFlow/index.tsx

```diff
@@ -51,70 +49,67 @@ function ConnectToNetSuiteButton({policyID, shouldDisconnectIntegrationBeforeCon
         },
     ];
 
-    return (
-        <>
-            <Button
-                onPress={() => {
-                    if (!isControlPolicy(policy)) {
-                        Navigation.navigate(ROUTES.WORKSPACE_UPGRADE.getRoute(policyID, CONST.UPGRADE_FEATURE_INTRO_MAPPING.netsuite.alias));
-                        return;
-                    }
+    useEffect(() => {
+        if (!isControlPolicy(policy)) {
+            Navigation.navigate(ROUTES.WORKSPACE_UPGRADE.getRoute(policyID, CONST.UPGRADE_FEATURE_INTRO_MAPPING.netsuite.alias));
```

Comment:
```suggestion
            Navigation.navigate(
                            ROUTES.WORKSPACE_UPGRADE.getRoute(policyID, CONST.UPGRADE_FEATURE_INTRO_MAPPING.netsuite.alias, ROUTES.POLICY_ACCOUNTING_NETSUITE_TOKEN_INPUT.getRoute(policyID)),
```

Reply:
Just to make sure we won't replicate the issue that we fixed here:
https://github.com/Expensify/App/pull/45730/files

Reply:
done :)